### PR TITLE
Set node env to production with --prod install.

### PR DESCRIPTION
**Pull request by samccone**
🐐  compat 🚀

### CODE REVIEW

**Comment by samccone** [comment]
Test failure is from flake

**Comment by kittens** [comment]
This pollutes the global environment. We should be setting `env.NODE_ENV` in `execute-lifecycle-script.js` and pass info to it from `config.executeLifecycleScript.js`.

**Comment by samccone** [comment]
sg

**Comment by samccone** [comment]
Ended up attaching the production bit to the config since that seems to make sense, and also since the config makes it all the way down to execute lifecycle script.

I did a version where I added a new arg to config.executeLifeCycle and then passed it through to execFromManifest and all the way down into the lifecycle execute and then set the flag there but... things got kinda messy. For that reason given the config that gets passed around I think this approach makes sense.

Now one thing that is interesting is that I now mutate the config object in the install phase which seems OK but a little different from just instantiating the config object with the option.

Given where the config object is built (prior to the flags being parsed) it seemed like that approach would also be odd.

Let me know what you think about my current approach and thanks for the 👓

File: src/cli/commands/install.js [hunk 1]
```diff
@@ -134,6 +134,8 @@ export class Install {
     this.config = config;
     this.flags = normalizeFlags(config, flags);
 
+    this.config.production = this.flags.production;
```
Comment (kittens) [inline src/cli/commands/install.js]: We can move that to `config.init` in `cli/index.js` and move the `--production` flag from `install.js` to `cli/index.js`.

Comment (kittens) [inline src/cli/commands/install.js]: #1517 will make this easier.

Comment (samccone) [inline src/cli/commands/install.js]: yes good call.

Comment (samccone) [inline src/cli/commands/install.js]: addressed :goat:

**Pull request merged**